web/rollup.config.js

```diff
@@ -49,5 +49,28 @@ export default {
         transformHtml: (html) => {
             return html.replace(/<script type=\"importmap\">[^<]+<\/script>/g, "");
         }
-    }), terser()],
+    }), terser(),
+    {
+        name: "inline-js",
+        closeBundle: () => {
+          const js = readFileSync("./app.component.js").toString();
```

Comment:
It is the bundled js file you need to load, and not app.component.js.

the bundle file named `app.[hash].js` contains all the javascript and html packed into one file, while `app.component.js` will try loading the other files which will not be available in the production deployment context.